avocado/plugins/diff.py

```diff
@@ -0,0 +1,387 @@
+
+    """
+    Implements the avocado 'diff' subcommand
+    """
+
+    name = 'diff'
+    description = 'Shows the difference between 2 jobs.'
+
+    def configure(self, parser):
+        """
+        Add the subparser for the diff action.
+
+        :param parser: Main test runner parser.
+        """
+        parser = super(Diff, self).configure(parser)
+
+        parser.add_argument("jobids",
+                            default=[], nargs=2,
+                            metavar="<JOB>",
+                            help='Provide two Jobs, identified by their Job '
+                            '(partial) ID or test results directory.')
+
+        parser.add_argument('--html', type=str,
+                            dest='html_output',
+                            metavar='FILE',
+                            help='Enable HTML output to the FILE where the '
+                            'result should be written. The value - (output to '
+                            'stdout) is not supported since not all HTML '
+                            'resources can be embedded into a single file '
+                            '(page resources will be copied to the output '
+                            'file dir)')
+
+        parser.add_argument('--open-browser',
+                            action='store_true',
+                            default=False,
+                            help='Generate and open a HTML report in your '
+                            'preferred browser. If no --html file is '
+                            'provided, we create a temporary file. '
+                            'Current: False.')
+
+        parser.add_argument('--diff-filter',
+                            dest='diff_filter',
+                            type=self._validate_filters,
+                            default=['all'],
+                            help='Diff filter: all,(-)cmdline,(-)time,'
+                            '(-)variants,(-)results,(-)config,(-)sysinfo '
+                            '(defaults to all).')
+
+        parser.add_argument('--paginator',
+                            choices=('on', 'off'), default='on',
+                            help='Turn the paginator on/off. '
+                            'Current: %(default)s')
+
+        self.term = output.TERM_SUPPORT
+
+    def run(self, args):
+        job1_dir, job1_id = self._setup_job(args.jobids[0])
+        job2_dir, job2_id = self._setup_job(args.jobids[1])
+
+        job1_data = self._get_job_data(job1_dir)
+        job2_data = self._get_job_data(job2_dir)
+
+        job1_results = []
+        job2_results = []
+
+        if (('all' in args.diff_filter or 'cmdline' in args.diff_filter) and
+           '-cmdline' not in args.diff_filter):
+
+            cmdline1 = self._get_command_line(job1_dir)
+            cmdline2 = self._get_command_line(job2_dir)
+
+            if self._has_diff([cmdline1], [cmdline2]):
+                command_line_header = ['\n', 'COMMAND LINE\n']
```

Comment:
Well then I'd at least use `rst` approach and use

```rst
COMMAND LINE
=============
```

And I'd put there the color, because at least I'm unable to distinguish where one section ends and next one begins. As for developers, they are used to:

```
/**********************************************
 * Section
 **********************************************/
```